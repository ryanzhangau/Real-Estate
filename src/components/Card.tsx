import React from 'react';

export interface Item {
  id: number;
  status: string;
  street: string;
  suburb: string;
  state: string;
  postcode: string;
  price: number;
  image: string;
}

interface CardProps {
  item: Item;
  key: number;
}

const Card: React.FC<CardProps> = (props: CardProps) => {
  const { item } = props;
  return (
    <div>
      <span>${item.price}</span>
      <span>{item.status}</span>
      <img src={item.image} alt={`${item.street} ${item.suburb}`} />
      <span>{item.street}</span>
      <span>{item.suburb}</span>
      <span>
        {item.state} {item.postcode}
      </span>
    </div>
  );
};

export default Card;
