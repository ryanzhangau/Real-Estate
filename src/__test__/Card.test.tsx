import React from 'react';
import { shallow } from '../enzyme';
import Card from '../components/Card';
import { ShallowWrapper } from 'enzyme';

describe('<Card />', () => {
  let component: ShallowWrapper;
  const item = {
    id: 1,
    status: 'current',
    street: '1 test street',
    suburb: 'Test',
    state: 'Victoria',
    postcode: '3000',
    price: 1200000,
    image: '/testimage.jpg'
  };
  beforeEach(() => {
    component = shallow(<Card item={item} key={1} />);
  });

  it('should render card', () => {
    expect(component).toMatchSnapshot();
  });
});
