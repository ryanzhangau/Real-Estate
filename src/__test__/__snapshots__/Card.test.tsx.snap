// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render card 1`] = `ShallowWrapper {}`;

exports[`card snapshot verion 1`] = `
<div className='Card_wrap'>
  <span className='Card_price'>$1200000</span>
  <span className='Card_status'>current</span>
  <span className='Card_featured-image'>
    <img src="/testimage.jpg" alt="1 test street Test" />
  </span>
  <span className='Card_street'>1 test street</span>
  <span className='Card_suburb'>Test</span>
  <span className='Card_state'>
    Victoria 3000
  </span>
</div>
`;
